File: Models/SpaceSystem.cs
using Microsoft.AspNetCore.Mvc;
using Newtonsoft.Json;
using System.Text.Json;
using System.Text.Json.Serialization;

namespace TestSystem.Models
{
 
    public class SpaceSystem
    {
        private List<SpaceObject> _spaceObjects = new();
        private string? _name = String.Empty;
        private ulong _age = 0;
        private SpaceObject? _centerOfGravity;
        public delegate void badSpaceObjectType(string message);
        public static event badSpaceObjectType? Notify;

        public SpaceSystem(string name, ulong age, SpaceObject? centerOfGravity , SpaceObject[]? planets)
        {
            name = name;
            age = age;
            if(centerOfGravity != null) CenterOfGravity = centerOfGravity; 
            if(planets != null) Planets = planets.ToList();
        }
        public SpaceSystem() { }

        public override string ToString()
        {
            return $"{Id} - {Name} - {Age}";
        }
        public int Id { get; set; }
        public string? Name 
        {
            get
            {
                return _name ?? string.Empty;
            }
            set
            { 
                _name = value; 
            }
        }
        public ulong Age    
        {
            get =>  _age;
            
            set 
            { 
                _age = value;
            }
        }
        public List<SpaceObject> Planets
        {
            get => _spaceObjects;
            set 
            { 
                _spaceObjects.AddRange(value.ToList());
            }
        }
        public SpaceObject? CenterOfGravity
        {
            get => _centerOfGravity;
            set 
            {
               switch (value?.Type)
                {
                    case "star":
                        _centerOfGravity = value;
                        break;
                    case "blachhole":
                        _centerOfGravity = value;
                        break;
                    default:
                        Notify?.Invoke("Wrong type of centerOfGravity");
                        break;
                }
            }
        }
    }
}

File: client/src/SpaceSystem.ts
import { SpaceObject } from "./SpaceObject";

export class SpaceSystem
{
    private SpaceObjects: SpaceObject[] = [];
    private Name: string = '';
    private Age = 0;
    private CenterOfGravity!: SpaceObject;
    public id!: number;

    constructor(name: string, age: number, centerOfGravity: SpaceObject, objects?: SpaceObject[]){
        this.Name = name;
        this.Age = age;
        this.CenterOfGravity = centerOfGravity;
        if(objects != null) this.SpaceObjects = objects;
    }

    public get name():string
        {
            return this.Name;
        }
        public set name(s: string)
        {
            this.Name = s;
        }

        public get age():number
        {
            return this.Age;
        }
        public set age(a: number)
        {
            this.Age = a;
        }

        public get centerOfGravity():SpaceObject
        {
            return this.CenterOfGravity;
        }
        public set centerOfGravity(s: SpaceObject)
        {
            this.CenterOfGravity = s;
        }

        public get spaceObjects():SpaceObject[]
        {
            return this.SpaceObjects;
        }
        public set CenterOspaceObjectsfGravity(s: SpaceObject[])
        {
            this.SpaceObjects = this.SpaceObjects.concat(s);
        }
}